libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts

```diff
@@ -180,4 +216,115 @@ export const mutations = {
     const response = await organizationApi.createGitToken(organizationId, gitTokenRequest)
     return response.data
   },
+  async createCloudProviderCredential({
+    organizationId,
+    cloudProvider,
+    credentialRequest,
+  }: {
+    organizationId: string
+    cloudProvider: CloudProviderEnum
+    credentialRequest: AwsCredentialsRequest | ScalewayCredentialsRequest | DoCredentialsRequest
+  }) {
```

Comment:
Use discriminated union to avoid `as` type cast.
eg.
https://github.com/Qovery/console/blob/staging/libs/domains/services/data-access/src/lib/domains-services-data-access.ts#L325

Reply:
I think I must add `as` for `AwsCredentialsRequest` because we use the enum `CloudProviderEnum` on the credential feature, and when I call the hook (ex:  https://github.com/Qovery/console/blob/feat/migrate-credentials-react-query-FRT-895/libs/shared/console-shared/src/lib/cluster-settings/feature/create-edit-credentials-modal-feature/create-edit-credentials-modal-feature.tsx). I need to refactor all these features if we want it, it should be better but not perfect  :/